src-v5/carousel.tsx

```diff
@@ -49,12 +49,20 @@ export const Carousel = (props: CarouselProps): React.ReactElement => {
     );
     to && props.beforeSlide(slide, endSlide);
     !props.disableAnimation && setAnimation(true);
-
     setCurrentSlide(to ?? currentSlide);
     setTimeout(
       () => {
         if (!isMounted.current) return;
-        to && props.afterSlide(currentSlide);
+
+        let nextIndex = to ?? currentSlide;
+        if (nextIndex < 0) {
+          nextIndex += count;
+        }
```

Comment:
if it's less than `0` that means we went backwards so we want to add the total count to the negative number to get the right index.